pypesto/petab/importer.py

```diff
@@ -78,6 +80,40 @@ def from_yaml(yaml_config: Union[dict, str],
             output_folder=output_folder,
             model_name=model_name)
 
+    def check_and_log_gradients(self, RTOL: float = 1e-2, ATOL: float = 1e-3):
+        """Check and log if gradients are compatible with gradient-based optimizers
+        Parameters
+        ----------
+        RTOL: relative error tolerance (default 1e-2)
+        ATOL: absolute error tolerance (default 1e-3)
+        """
+        par = np.asarray(self.petab_problem.x_nominal_scaled)
+        problem = self.create_problem()
+        objective = problem.objective
+        objective.amici_solver.setSensitivityMethod(
+            amici.SensitivityMethod_forward)
+        objective.amici_solver.setAbsoluteTolerance(1e-10)
+        objective.amici_solver.setRelativeTolerance(1e-12)
+        free_indices = par[problem.x_free_indices]
+        dfs = []
+        dfs.append(objective.check_grad_multi_eps(
+                    free_indices,
+                    multi_eps=[1e-3, 1e-4, 1e-5]))
+
+        modes = [MODE_FUN, MODE_RES]
+        for mode in modes:
+            try:
+                dfs.append(objective.check_grad(free_indices, mode=mode))
+            except (RuntimeError, ValueError):
+                dfs.append(dfs[0])
```

Comment:
```suggestion
        dfs = []

        modes = [MODE_FUN, MODE_RES]
        for mode in modes:
            try:
                dfs.append(objective.check_grad_multi_eps(
                            free_indices,
                            multi_eps=[1e-3, 1e-4, 1e-5],
                            mode = mode))
            except (RuntimeError, ValueError):
                dfs.append(dfs[0])
```
I think you could just leave out `check_grad`.